compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -4083,7 +4062,7 @@ OMR::Z::CodeGenerator::gprClobberEvaluate(TR::Node * node, bool force_copy, bool
          return targetRegister;
          }
       }
-   return self()->evaluate(node);
+   return srcRegister;
```

Comment:
Can you explain why it's ok to no longer evaluate the node and instead just return `srcRegister`?

Reply:
The `node` has already been evaluated at the start of the evaluator. Calling `evaluate` on it again will simply return the same register as before, so let's save ourselves the function call and return it directly.